monitor.py

```diff
@@ -35,8 +35,24 @@ def fetch_release_date(release):
                 if p_ele:
                     if "Created:" in p_ele:
                         start_date = p_ele.split(" ")[1]+" "+p_ele.split(" ")[2]
-                        break
-            return start_date
+                        return start_date
+            form = soup.find("form", {"method": "GET"})
+            if form:
+                a_tag = form.find("a", href=True)
+                if a_tag and "changelog" in a_tag["href"]:
+                    changelog_url = constants.RELEASE_BASE_URL + a_tag["href"]
+                    changelog_resp = requests.get(changelog_url, verify=False, timeout=15)
+                    if changelog_resp.status_code == 200:
+                        lines = changelog_resp.text.splitlines()
+                        for line in lines:
+                            if "Created:" in line:
+                                parts = line.split()
+                                if len(parts) >= 3:
```

Comment:
Why do we need this check and Can we move the common code lines out of this if block